src/ImageSharp.Web/Commands/CommandCollection.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace SixLabors.ImageSharp.Web.Commands
+{
+    /// <summary>
+    /// Represents an ordered collection of processing commands.
+    /// </summary>
+    public sealed class CommandCollection : KeyedCollection<string, KeyValuePair<string, string>>
+    {
+        private readonly List<string> keys = new();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CommandCollection"/> class.
+        /// </summary>
+        public CommandCollection()
+            : this(StringComparer.OrdinalIgnoreCase)
+        {
+        }
+
+        private CommandCollection(IEqualityComparer<string> comparer)
+            : base(comparer)
```

Comment:
Keep in mind that this constructor already creates an internal lookup dictionary: https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.keyedcollection-2.-ctor?view=net-6.0#system-collections-objectmodel-keyedcollection-2-ctor(system-collections-generic-iequalitycomparer((-0))-system-int32).

Reply:
I considered implementing `IDictionary<string, string>` but it leads to a confusing API since the type inherits `Collection<T>`. `Contains` vs `ContainsKey` for example. There's also virtualization issues passing around _IDictionary<T>_ all the time. 

I'm not quite sure what you're pointing out re the constructor. Are you suggesting using the internal dictionary to implement the interface or for something else?

Reply:
Confusion can be avoided by using an explicit interface implementation (which might already be required for some conflicting members)...

The point I wanted to make, is that this implementation stores the items both in the collection and internal lookup dictionary, in addition to keeping a sorted list of keys.

Too bad `OrderedDictionary` isn't using generics, although I see there is an [internal MS implementation](https://referencesource.microsoft.com/#System.ServiceModel.Internals/System/Runtime/Collections/OrderedDictionary.cs).